src/frontend/src/flows/addDevice/manage/pollForTentativeDevice.ts

```diff
@@ -1,61 +1,99 @@
 import { html } from "lit-html";
 import { asyncReplace } from "lit-html/directives/async-replace.js";
+import { createRef, ref, Ref } from "lit-html/directives/ref.js";
+import type QrCreator from "qr-creator"; // XXX: import to only import the _type_ to avoid pulling in the whole module (module itself is used as a dynamic import)
 import {
   DeviceData,
   Timestamp,
 } from "../../../../generated/internet_identity_types";
+import { checkmarkIcon, copyIcon } from "../../../components/icons";
 import { mainWindow } from "../../../components/mainWindow";
+import { toast } from "../../../components/toast";
 import { addDeviceLink } from "../../../utils/addDeviceLink";
 import { AsyncCountdown } from "../../../utils/countdown";
 import { AuthenticatedConnection } from "../../../utils/iiConnection";
-import { renderPage } from "../../../utils/lit-html";
+import { mount, renderPage, withRef } from "../../../utils/lit-html";
 import { delayMillis } from "../../../utils/utils";
 
 const pollForTentativeDeviceTemplate = ({
+  userNumber,
   cancel,
   remaining,
-  link,
+  origin,
 }: {
+  userNumber: bigint;
   cancel: () => void;
   remaining: AsyncIterable<string>;
-  link: string;
+  origin: string;
 }) => {
+  const link = addDeviceLink({ userNumber, origin });
+  const copyLink_ = () => navigator.clipboard.writeText(link);
+
+  const linkCopyElement: Ref<HTMLElement> = createRef();
+  // Copy the link and give visual feedback on success
+  const copyLink = async () => {
+    try {
+      await copyLink_();
+      withRef(linkCopyElement, (linkCopyElement) => {
+        linkCopyElement.classList.add("is-copied");
+      });
+    } catch (e: unknown) {
+      toast.error("Could not copy link to clipboard");
+      console.error("Could not copy link to clipboard", e);
+    }
+  };
+
   const pageContentSlot = html`
+    <p class="t-paragraph">Identity Anchor ${userNumber}</p>
     <hgroup>
-      <h1 class="t-title t-title--main">Add a Trusted Device</h1>
-      <p class="t-lead">
-        Complete the steps below
-        <strong class="t-string"> on the device you want to add: </strong>
-      </p>
+      <h1 class="t-title t-title--main">Add Trusted Device</h1>
+      <p class="t-lead">Follow these steps on your new device</p>
     </hgroup>
+    <div
+      class="t-centered c-qrcode l-stack"
+      ${mount((container) =>
+        container instanceof HTMLElement
+          ? displayQR({ link, container })
+          : undefined
+      )}
+    ></div>
+    <p data-role="add-device-link" class="t-paragraph l-stack t-wrap">
+      ${link}
+    </p>
+    <!-- this is an absolute hack: position relative is needed for c-button__icon to be anchored, and
```

Comment:
@meodai I tried reusing `c-button__icon` but in the end I don't think that's a very good idea. What do you think?

Reply:
Not sure what you are trying to achieve. Can you give be a bit of context?